### PR TITLE
test: Add Tests and Improve Test Coverage for packages/web3/src/contract

### PR DESCRIPTION
**Pull request by martinvibes**
test: add test for non tested files and improved test converage #453

### CODE REVIEW

**Comment by martinvibes** [comment]
hey @polarker this test really stressed me out 
but learnt alot :)

File: packages/web3/src/contract/events.ts [hunk 1]
```diff
@@ -1,24 +1,8 @@
-/*
```
Comment (h0ngcha0) [inline packages/web3/src/contract/events.ts:1]: Q: Why do we need to change this file?

**Comment by h0ngcha0** [comment]
Hey, I can not build and run tests with the branch, could you please make sure that it builds and all tests are executed correctly?

```bash
pnpm install
npm run build
npm run test
```

**Comment by martinvibes** [comment]
hey @h0ngcha0 @polarker it is building and passing successfully too on my end 

image

File: packages/web3/src/contract/events.ts (hunk 1 above)
Comment (h0ngcha0) [reply packages/web3/src/contract/events.ts:1]: Unresolve this issue since it is not resolved.

**Comment by h0ngcha0** [comment]
> hey @h0ngcha0 @polarker it is building and passing successfully too on my end
> 
> image

It doesn't build for me:

image

**Comment by martinvibes** [comment]
You have to cd into packages/web3 then build

File: packages/web3/src/contract/events.ts (hunk 1 above)
Comment (martinvibes) [reply packages/web3/src/contract/events.ts:1]: It doesn't test that's why 😔

**Comment by h0ngcha0** [comment]
> You have to cd into packages/web3 then build

It should be able to build at the top directory, that is how CI runs the pipeline as well

**Comment by martinvibes** [comment]
Hey @h0ngcha0 running tests from root shows me error from files that doesn't have anything to do with the tests and files i didn't touch

**Comment by martinvibes** [comment]
The tests are literally passing and building which is good :)

**Comment by h0ngcha0** [comment]
> Hey @h0ngcha0 running tests from root shows me error from files that doesn't have anything to do with the tests and files i didn't touch

The build error was not present in the master branch, so it is introduced by this PR. 
Sorry, I will not look into the code unless it can build.

**Comment by martinvibes** [comment]
hey @h0ngcha0 i opened a new PR and all tests are passing and builds successful #479

**Pull request closed without merging**